adapter-base/src/main/java/org/eclipse/hono/adapter/AbstractProtocolAdapterBase.java

```diff
@@ -1071,8 +1087,8 @@ public void registerLivenessChecks(final HealthCheckHandler handler) {
         if (commandRouterClient instanceof ServiceClient) {
             ((ServiceClient) commandRouterClient).registerLivenessChecks(handler);
         }
-        if (telemetrySender instanceof ServiceClient) {
-            ((ServiceClient) telemetrySender).registerLivenessChecks(handler);
+        if (telemetrySenders.getAmqpSender() instanceof ServiceClient) {
+            ((ServiceClient) telemetrySenders.getAmqpSender()).registerLivenessChecks(handler);
```

Comment:
IMHO we should encapsulate these checks and the setting of the liveness/readiness checks within the MultiMessagingSenders class ...

Reply:
I added this in my last commit. Please have a look.